google/cloud/bigquery/client.py

```diff
@@ -2838,6 +2871,7 @@ def _do_resumable_upload(
         num_retries: int,
         timeout: Optional[ResumableTimeoutType],
         project: Optional[str] = None,
+        checksum: Optional[str] = None,
```

Comment:
The intention here is to have a default for the public method, but not the private methods, correct?

Reply:
yes. that is what I was thinking.

I looked into this and this will require probably some changes to the google media library and likely the backend.